python/lsst/meas/extensions/piff/piffPsfDeterminer.py

```diff
@@ -0,0 +1,275 @@
+    """Rescale weight of bright stars to cap the computed SNR.
+
+    Parameters
+    ----------
+    imArr : `ndarray`
+        Signal (image) array of stamp.
+    weightArr : `ndarray`
+        Weight map array.  May be rescaled in place.
+    good : `ndarray`
+        Index array of pixels to use when computing SNR.
+    maxSNR : `float`
+        Threshold for adjusting variance plane implementing maximum SNR.
+    """
+    F = np.sum(weightArr[good]*imArr[good]**2, dtype=float)
+    Npix = np.sum(good)
+    SNR = 0.0 if F < Npix else (F-Npix)/np.sqrt(F)
+    # rescale weight of bright stars.  Essentially makes an error floor.
+    if SNR > maxSNR:
+        factor = (maxSNR / SNR)**2
+        weightArr[good] *= factor
+
+
+def _computeWeightAlternative(maskedImage, maxSNR):
+    """Alternative algorithm for creating weight map.
+
+    This version is equivalent to that used by Piff internally.  The weight map
+    it produces tends to leave a residual when removing the Poisson component
+    due to the signal.  We leave it here as a reference, but without intending
+    that it be used.
+    """
+    imArr = maskedImage.image.array
+    varArr = maskedImage.variance.array
+    good = (varArr != 0) & np.isfinite(varArr) & np.isfinite(imArr)
+
+    fit = np.polyfit(imArr[good], varArr[good], deg=1)
+    # fit is [1/gain, sky_var]
+    gain = 1./fit[0]
+    varArr[good] -= imArr[good] / gain
+    weightArr = np.zeros_like(imArr, dtype=float)
+    weightArr[good] = 1./varArr[good]
+
+    applyMaxSNR(imArr, weightArr, good, maxSNR)
+    return weightArr
+
+
+class PiffPsfDeterminerTask(BasePsfDeterminerTask):
+    """A measurePsfTask psf estimator using Piff as the implementation.
+    """
+    ConfigClass = PiffPsfDeterminerConfig
+
+    def determinePsf(
+        self, exposure, psfCandidateList, metadata=None, flagKey=None
+    ):
+        """Determine a Piff PSF model for an exposure given a list of PSF
+        candidates.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+           Exposure containing the psf candidates.
+        psfCandidateList : `list` of `lsst.meas.algorithms.PsfCandidate`
+           A sequence of PSF candidates typically obtained by detecting sources
+           and then running them through a star selector.
+        metadata : `lsst.daf.base import PropertyList` or `None`, optional
+           A home for interesting tidbits of information.
+        flagKey : `str`, optional
+           Schema key used to mark sources actually used in PSF determination.
+
+        Returns
+        -------
+        psf : `lsst.meas.extensions.piff.PiffPsf`
+           The measured PSF.
+        psfCellSet : `lsst.afw.math.SpatialCellSet`
+           The PSF candidates.
+        """
+        emi = exposure.getMaskedImage()
+        ebbox = emi.getBBox(afwImage.PARENT)
+        psfCellSet = afwMath.SpatialCellSet(
+            ebbox, self.config.sizeCellX, self.config.sizeCellY
+        )
```

Comment:
It confuses me too, and I included it really only because it's one of the return values defined by `BasePsfDeterminerTask`.  It might be worth asking on #dm-psf or another slack channel whether this is worth keeping.

Reply:
Actually, looks like the cellSet is used more extensively at least in `pcaPsfDeterminer`, so there's at least that use case.  It might be better to just ignore it here though, similar to it's mostly ignored for PSFEx.

Reply:
I just went through and excised the cellSet stuff completely since it wasn't being used.  A cellSet is still part of the determiner return API, of course, but that's just hard-coded to None now (and documented as such).  I _think_ that's clearer.